src/compiler/checker.ts

```diff
@@ -21705,6 +21705,10 @@ namespace ts {
                 case SyntaxKind.NonNullExpression:
                 case SyntaxKind.ParenthesizedExpression:
                     return getFlowCacheKey((<NonNullExpression | ParenthesizedExpression>node).expression, declaredType, initialType, flowContainer);
+                case SyntaxKind.QualifiedName:
+                    const left = getFlowCacheKey((<QualifiedName>node).left, declaredType, initialType, flowContainer);
+                    const right = (<QualifiedName>node).right.escapedText;
+                    return `${left}.${right}`;

```

Comment:
This should use the same structure as the property/element access code below. Specifically, `getFlowCacheKey` can return `undefined`.

The way that `isMatchingReference` looks is probably the right way to copy the code, but it might also be possible to unify the handling of all three node kinds into one code path by finding/making more functions like getAccessedPropertyName.

Reply:
🤔 I think I gonna just copy from `isMatchingReference` because the structure of `QualifiedName` is kinda different from `ElementAccessExpression`